Update readme docs for Go SDK (#56975)
mistercrunch/Airflow: go-sdk/README.md

 under the License.
 -->

# Apache Airflow Go Task SDK

The Go SDK uses the Task Execution Interface (TEI or Task API) introduced in AIP-72 with Airflow 3.0.0 to give
Task functions written in Go full access to the Airflow "model", natively in go.

The Task API however does not provide a means to get the `ExecuteTaskWorkload` to the go worker itself. For

> that we use the Edge Executor API

Author: ```suggestion
that we use the Edge Executor API.
```

> Longer term we will likely need to stabilize the Edge Executor API and write a go client for that.

Reviewer: ```suggestion
that we use the Edge Executor API.
Longer term we will likely need to stabilize the Edge Executor API and add versioning to it.
```

> Longer term we will likely need to stabilize the Edge Executor API and write a go client for that.

Author: ```suggestion
Longer term we will likely need to stabilize the Edge Executor API and add versioning to it.
```

> Since Go is a compiled language (putting aside projects such as [YAEGI](https://github.com/traefik/yaegi) that allow go to be interpreted) all tasks must be a) compiled in to the binary, and b) "registered" inside the worker process in order to be executed.

Reviewer 2: ```suggestion
Since Go is a compiled language (putting aside projects such as [YAEGI](https://github.com/traefik/yaegi) that allow Go to be interpreted), all tasks must:

1. Be compiled into a binary ahead of time, and  
2. Be registered inside the worker process in order to be executed.
```

> [!NOTE]
> This Golang SDK is under active development and is not ready for prime-time yet.

## Quickstart

- See [`example/bundle/main.go`](./example/bundle/main.go) for an example dag bundle where you can define your task functions

> - Compile this in to a binary:

Reviewer 2: ```suggestion
- Compile this into a binary:
```
